crates/hyperqueue/src/stream/reader/logfile.rs

```diff
@@ -181,6 +191,64 @@ impl LogFile {
         }
     }
 
+    fn _gather_infos<'a>(
+        index: &'a BTreeMap<JobTaskId, TaskInfo>,
+        tasks: &Option<IntArray>,
+    ) -> anyhow::Result<Vec<(JobTaskId, &'a InstanceInfo)>> {
+        Ok(match tasks {
+            Some(ref array) => {
+                let mut infos = Vec::new();
+                for task_id in array.iter() {
+                    if let Some(task_info) = index.get(&JobTaskId::new(task_id)) {
+                        infos.push((JobTaskId::new(task_id), task_info.last_instance()));
+                    } else {
+                        anyhow::bail!("Task {} not found", task_id);
+                    }
+                }
+                infos
+            }
+            None => index
+                .iter()
+                .map(|(&task_id, task_info)| (task_id, task_info.last_instance()))
+                .collect(),
+        })
+    }
+
+    pub fn export(&mut self, opts: &ExportOpts) -> anyhow::Result<()> {
+        let task_infos = Self::_gather_infos(&self.index, &opts.task)?;
+        let mut result = Vec::new();
+        self.file.seek(SeekFrom::Start(self.start_pos))?;
+        let mut last_pos: i64 = self.start_pos as i64;
+        let mut buffer = Vec::new();
+        for (task_id, instance) in &task_infos {
```

Comment:
This seems to be duplicated with stream reading in `cat`. It would be nice to create an `Iterator` over the chunks.